packages/navie/src/commands/review-command.ts

```diff
@@ -0,0 +1,138 @@
+
+Provide your feedback with specific references to the code snippets in question. Offer clear, actionable suggestions for improvements, and maintain a polite and professional tone throughout your review.`;
+
+const CHAIN_OF_THOUGHT = `Before presenting your review to the user, internally perform a detailed chain-of-thought analysis of the changes, focusing on the specified areas of concern. Use this internal reasoning to thoroughly understand the implications of the changes and to inform your feedback.
+
+Your task is to analyze the changes based on your internal reasoning and provide constructive, professional feedback to the user. Focus on the areas highlighted by the user, but also mention any critical issues you discover in other aspects. Provide your feedback with specific references to the code snippets in question. Offer clear, actionable suggestions for improvements, and maintain a polite and professional tone throughout your review.`;
+
+const REQUEST_FINAL_SUBMISSION = `Now, provide your feedback in a summarized form. Offer clear, actionable suggestions for improvements, and maintain a polite and professional tone throughout your review. Please do not include any analysis that was not conclusive based on your internal reasoning. Your full response will be added to internal documentation, so avoid additional chatter (e.g., "if you need further assistance", "certainly", etc.).`;
+
+export default class ReviewCommand implements Command {
+  private static readonly SearchTermRegex = /[\w\d]{2,}/g;
+
+  constructor(
+    private readonly options: ExplainOptions,
+    private readonly completionService: CompletionService,
+    private readonly lookupContextService: LookupContextService
+  ) {}
+
+  async *execute(req: CommandRequest): AsyncIterable<string> {
+    if (!req.codeSelection) {
+      // This should never happen.
+      yield 'An unknown error occurred. Please try again.';
+      return;
+    }
+
+    if (typeof req.codeSelection === 'string') {
+      yield 'Your `appmap` binary is out of date. Please update your client to the latest version.';
+      yield '';
+      yield 'Updates typically take place automatically. You can try restarting your code editor, or follow the manual instructions here:';
+      yield 'https://appmap.io/docs/reference/appmap-airgapped-install.html#download-the-appmap-application-binaries';
+      return;
+    }
+
+    const gitDiff = req.codeSelection.find(
+      (cs): cs is UserContext.CodeSnippetItem =>
+        cs.type === 'code-snippet' && cs.location === 'git diff'
```

Comment:
Let's define `git diff` as a constant since it's being used in this structured way. 

Is it going to cause issues in the UI if `git diff` is interpreted as a location? Should we add a new kind of UserContext item type?